bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/IProgressMonitor.java

```diff
@@ -253,6 +253,37 @@ public default IProgressMonitor slice(int work) {
 		return new SlicedProgressMonitor(this, work);
 	}
 
+	/**
+	 * Creates a new IProgressMonitor that will consume the given number of ticks
+	 * from this IProgressMonitor.
```

Comment:
I think we should add some note here about what the difference to slice (e.g. usually can not be used in MT environments) and if it is safe to call begin task on the new instance.

Reply:
i will add something